python/ray/autoscaler/azure/example-full.yaml

```diff
@@ -132,14 +132,20 @@ setup_commands:
     # Note: if you're developing Ray, you probably want to create an AMI that
     # has your Ray repo pre-cloned. Then, you can replace the pip installs
     # below with a git checkout <your_sha> (and possibly a recompile).
+    # - echo 'conda activate py37_pytorch' >> ~/.bashrc'
+    - echo 'conda activate py37_tensorflow' >> ~/.bashrc'
     - pip install -U https://raywheels.blob.core.windows.net/ray/ray-0.9.0.dev0-cp37-cp37m-manylinux1_x86_64.whl
     # Consider uncommenting these if you also want to run apt-get commands during setup
     # - sudo pkill -9 apt-get || true
     # - sudo pkill -9 dpkg || true
     # - sudo dpkg --configure -a
 
 # Custom commands that will be run on the head node after common setup.
-head_setup_commands: []
+head_setup_commands:
+    - >
+        pip install
+        azure-cli-core azure-core azure-mgmt-authorization azure-mgmt-network azure-mgmt-compute azure-mgmt-msi
+        paramiko aiohttp grpcio psutil setproctitle
```

Comment:
we should check to see which libraries must be specified, i think some of these are in setup.py, others are dependencies of specified packages, so _should_ be pulled in automatically?

Reply:
removed aiohttp and grpcio. running pipdeptree now

Reply:
we could remove paramiko since azure-cli-core depends on it, but we use it directly I'd keep it, just in case azure-cli-core removes the dependency in the future? the rest seems not dependent

pipdeptree -p azure-cli-core,azure-core,azure-mgmt-authorization,azure-mgmt-network,azure-mgmt-compute,azure-mgmt-msi

azure-cli-core==2.1.0
  - adal [required: ~=1.2, installed: 1.2.2]
    - cryptography [required: >=1.1.0, installed: 2.8]
      - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
        - pycparser [required: Any, installed: 2.19]
      - six [required: >=1.4.1, installed: 1.14.0]
    - PyJWT [required: >=1.0.0, installed: 1.7.1]
    - python-dateutil [required: >=2.1.0, installed: 2.8.1]
      - six [required: >=1.5, installed: 1.14.0]
    - requests [required: >=2.0.0, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - antlr4-python3-runtime [required: Any, installed: 4.8]
  - argcomplete [required: ~=1.8, installed: 1.11.1]
    - importlib-metadata [required: >=0.23,<2, installed: 1.5.0]
      - zipp [required: >=0.5, installed: 2.1.0]
  - azure-cli-nspkg [required: >=2.0.0, installed: 3.0.4]
    - azure-nspkg [required: >=2.0.0, installed: 3.0.2]
  - azure-cli-telemetry [required: Any, installed: 1.0.4]
    - applicationinsights [required: >=0.11.1,<0.12, installed: 0.11.9]
    - azure-cli-nspkg [required: >=2.0.0, installed: 3.0.4]
      - azure-nspkg [required: >=2.0.0, installed: 3.0.2]
    - portalocker [required: ~=1.2, installed: 1.5.2]
  - azure-mgmt-resource [required: ~=8.0.1, installed: 2.2.0]
    - azure-common [required: ~=1.1, installed: 1.1.24]
    - msrest [required: >=0.5.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrestazure [required: >=0.4.32,<2.0.0, installed: 0.6.2]
      - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
        - cryptography [required: >=1.1.0, installed: 2.8]
          - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
            - pycparser [required: Any, installed: 2.19]
          - six [required: >=1.4.1, installed: 1.14.0]
        - PyJWT [required: >=1.0.0, installed: 1.7.1]
        - python-dateutil [required: >=2.1.0, installed: 2.8.1]
          - six [required: >=1.5, installed: 1.14.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - isodate [required: >=0.6.0, installed: 0.6.0]
          - six [required: Any, installed: 1.14.0]
        - requests [required: ~=2.16, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
        - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
          - oauthlib [required: >=3.0.0, installed: 3.1.0]
          - requests [required: >=2.0.0, installed: 2.22.0]
            - certifi [required: >=2017.4.17, installed: 2019.11.28]
            - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
            - idna [required: >=2.5,<2.9, installed: 2.8]
            - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - colorama [required: >=0.3.9, installed: 0.4.3]
  - humanfriendly [required: ~=4.7, installed: 4.18]
  - jmespath [required: Any, installed: 0.9.4]
  - knack [required: ~=0.6.2, installed: 0.6.3]
    - argcomplete [required: Any, installed: 1.11.1]
      - importlib-metadata [required: >=0.23,<2, installed: 1.5.0]
        - zipp [required: >=0.5, installed: 2.1.0]
    - colorama [required: Any, installed: 0.4.3]
    - jmespath [required: Any, installed: 0.9.4]
    - pygments [required: Any, installed: 2.5.2]
    - pyyaml [required: Any, installed: 5.3]
    - six [required: Any, installed: 1.14.0]
    - tabulate [required: Any, installed: 0.8.6]
  - msrest [required: >=0.4.4, installed: 0.6.11]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - isodate [required: >=0.6.0, installed: 0.6.0]
      - six [required: Any, installed: 1.14.0]
    - requests [required: ~=2.16, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
      - oauthlib [required: >=3.0.0, installed: 3.1.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - msrestazure [required: >=0.6.2, installed: 0.6.2]
    - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
      - cryptography [required: >=1.1.0, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]
      - PyJWT [required: >=1.0.0, installed: 1.7.1]
      - python-dateutil [required: >=2.1.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - paramiko [required: >=2.0.8,<3.0.0, installed: 2.7.1]
    - bcrypt [required: >=3.1.3, installed: 3.1.7]
      - cffi [required: >=1.1, installed: 1.13.2]
        - pycparser [required: Any, installed: 2.19]
      - six [required: >=1.4.1, installed: 1.14.0]
    - cryptography [required: >=2.5, installed: 2.8]
      - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
        - pycparser [required: Any, installed: 2.19]
      - six [required: >=1.4.1, installed: 1.14.0]
    - pynacl [required: >=1.0.1, installed: 1.3.0]
      - cffi [required: >=1.4.1, installed: 1.13.2]
        - pycparser [required: Any, installed: 2.19]
      - six [required: Any, installed: 1.14.0]
  - PyJWT [required: Any, installed: 1.7.1]
  - pyopenssl [required: >=17.1.0, installed: 19.1.0]
    - cryptography [required: >=2.8, installed: 2.8]
      - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
        - pycparser [required: Any, installed: 2.19]
      - six [required: >=1.4.1, installed: 1.14.0]
    - six [required: >=1.5.2, installed: 1.14.0]
  - pyyaml [required: Any, installed: 5.3]
  - requests [required: ~=2.20, installed: 2.22.0]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
    - idna [required: >=2.5,<2.9, installed: 2.8]
    - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - six [required: ~=1.12, installed: 1.14.0]
  - wheel [required: ==0.30.0, installed: 0.30.0]
azure-core==1.2.2
  - requests [required: >=2.18.4, installed: 2.22.0]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
    - idna [required: >=2.5,<2.9, installed: 2.8]
    - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - six [required: >=1.6, installed: 1.14.0]
azure-mgmt-authorization==0.50.0
  - azure-common [required: ~=1.1,>=1.1.12, installed: 1.1.24]
  - azure-mgmt-nspkg [required: >=2.0.0, installed: 3.0.2]
    - azure-nspkg [required: >=3.0.0, installed: 3.0.2]
  - msrestazure [required: >=0.4.27,<2.0.0, installed: 0.6.2]
    - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
      - cryptography [required: >=1.1.0, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]
      - PyJWT [required: >=1.0.0, installed: 1.7.1]
      - python-dateutil [required: >=2.1.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
azure-mgmt-compute==4.6.2
  - azure-common [required: ~=1.1, installed: 1.1.24]
  - msrest [required: >=0.5.0, installed: 0.6.11]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - isodate [required: >=0.6.0, installed: 0.6.0]
      - six [required: Any, installed: 1.14.0]
    - requests [required: ~=2.16, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
      - oauthlib [required: >=3.0.0, installed: 3.1.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - msrestazure [required: >=0.4.32,<2.0.0, installed: 0.6.2]
    - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
      - cryptography [required: >=1.1.0, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]
      - PyJWT [required: >=1.0.0, installed: 1.7.1]
      - python-dateutil [required: >=2.1.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
azure-mgmt-msi==0.2.0
  - azure-common [required: ~=1.1, installed: 1.1.24]
  - azure-mgmt-nspkg [required: >=2.0.0, installed: 3.0.2]
    - azure-nspkg [required: >=3.0.0, installed: 3.0.2]
  - msrestazure [required: >=0.4.27,<2.0.0, installed: 0.6.2]
    - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
      - cryptography [required: >=1.1.0, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]
      - PyJWT [required: >=1.0.0, installed: 1.7.1]
      - python-dateutil [required: >=2.1.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
azure-mgmt-network==2.7.0
  - azure-common [required: ~=1.1, installed: 1.1.24]
  - msrest [required: >=0.5.0, installed: 0.6.11]
    - certifi [required: >=2017.4.17, installed: 2019.11.28]
    - isodate [required: >=0.6.0, installed: 0.6.0]
      - six [required: Any, installed: 1.14.0]
    - requests [required: ~=2.16, installed: 2.22.0]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
      - idna [required: >=2.5,<2.9, installed: 2.8]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
      - oauthlib [required: >=3.0.0, installed: 3.1.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
  - msrestazure [required: >=0.4.32,<2.0.0, installed: 0.6.2]
    - adal [required: >=0.6.0,<2.0.0, installed: 1.2.2]
      - cryptography [required: >=1.1.0, installed: 2.8]
        - cffi [required: >=1.8,!=1.11.3, installed: 1.13.2]
          - pycparser [required: Any, installed: 2.19]
        - six [required: >=1.4.1, installed: 1.14.0]
      - PyJWT [required: >=1.0.0, installed: 1.7.1]
      - python-dateutil [required: >=2.1.0, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - requests [required: >=2.0.0, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
    - msrest [required: >=0.6.0,<2.0.0, installed: 0.6.11]
      - certifi [required: >=2017.4.17, installed: 2019.11.28]
      - isodate [required: >=0.6.0, installed: 0.6.0]
        - six [required: Any, installed: 1.14.0]
      - requests [required: ~=2.16, installed: 2.22.0]
        - certifi [required: >=2017.4.17, installed: 2019.11.28]
        - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
        - idna [required: >=2.5,<2.9, installed: 2.8]
        - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]
      - requests-oauthlib [required: >=0.5.0, installed: 1.3.0]
        - oauthlib [required: >=3.0.0, installed: 3.1.0]
        - requests [required: >=2.0.0, installed: 2.22.0]
          - certifi [required: >=2017.4.17, installed: 2019.11.28]
          - chardet [required: >=3.0.2,<3.1.0, installed: 3.0.4]
          - idna [required: >=2.5,<2.9, installed: 2.8]
          - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.8]

Reply:
yeah, makes sense I think psutil and setproctitle are vendored in the third-party folder during building the wheel.

let's stick with: azure-cli-core azure-core azure-mgmt-authorization azure-mgmt-compute azure-mgmt-network  azure-mgmt-msi paramiko